pkg/cloudprovider/provider/anexia/provider_test.go

```diff
@@ -150,28 +159,37 @@ func TestAnexiaProvider(t *testing.T) {
 		})
 
 		providerStatus := anxtypes.ProviderStatus{}
-		ctx := utils.CreateReconcileContext(context.Background(), utils.ReconcileContext{
```

Comment:
we should have more test cases, as I read this we replace the one successful test (using `TemplateID`) with one using a named Template instead, but only checking it does not return an error.

We should also test using `TemplateID` and using `Template`+`TemplateBuild` and make sure it's resolved to the correct template IDs in all those cases.